client/settings/general-settings/index.js

```diff
@@ -9,50 +9,24 @@ import interpolateComponents from '@automattic/interpolate-components';
 /**
  * Internal dependencies
  */
-import { useDevMode, useIsWCPayEnabled, useTestMode } from 'wcpay/data';
+import { useDevMode, useTestMode } from 'wcpay/data';
 import CardBody from '../card-body';
 import InlineNotice from 'wcpay/components/inline-notice';
 import SetupLivePaymentsModal from 'wcpay/overview/modal/setup-live-payments';
 import TestModeConfirmationModal from './test-mode-confirm-modal';
-import { recordEvent } from 'tracks';
+import EnableWooPaymentsCheckbox from './enable-woopayments-checkbox';
 
 const GeneralSettings = () => {
-	const [ isWCPayEnabled, setIsWCPayEnabled ] = useIsWCPayEnabled();
 	const [ isEnabled, updateIsTestModeEnabled ] = useTestMode();
 	const [ modalVisible, setModalVisible ] = useState( false );
 	const isDevModeEnabled = useDevMode();
 	const [ testModeModalVisible, setTestModeModalVisible ] = useState( false );
 
-	const handleWcpayEnabledChange = ( enableWCPay ) => {
-		setIsWCPayEnabled( enableWCPay );
-
-		recordEvent( 'wcpay_gateway_toggle', {
-			action: enableWCPay ? 'enable' : 'disable',
-			context: 'wcpay-settings',
-		} );
-	};
-
 	return (
 		<>
 			<Card>
 				<CardBody>
-					<CheckboxControl
```

Comment:
I encapsulated this checkbox (and the confirmation dialog) in a new component, to make it easier to test.